lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+#include "SDIR/Translate/Translation.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/IR/AsmState.h"
+
+using namespace mlir;
+using namespace sdir;
+
+//===----------------------------------------------------------------------===//
+// TranslateToSDFG
+//===----------------------------------------------------------------------===//
+
+// NOTE: Prefer calling the specific translating functions instead
+LogicalResult translateToSDFG(Operation &op, JsonEmitter &jemit) {
+  if (SDFGNode node = dyn_cast<SDFGNode>(op))
+    return translateSDFGToSDFG(node, jemit);
+
+  if (StateNode node = dyn_cast<StateNode>(op))
+    return translateStateToSDFG(node, jemit);
+
+  if (TaskletNode node = dyn_cast<TaskletNode>(op))
+    return translateTaskletToSDFG(node, jemit);
+
+  if (MapNode node = dyn_cast<MapNode>(op))
+    return translateMapToSDFG(node, jemit);
+
+  if (ConsumeNode node = dyn_cast<ConsumeNode>(op))
+    return translateConsumeToSDFG(node, jemit);
+
+  if (EdgeOp Op = dyn_cast<EdgeOp>(op))
+    return translateEdgeToSDFG(Op, jemit);
+
+  if (AllocOp Op = dyn_cast<AllocOp>(op))
+    return translateAllocToSDFG(Op, jemit);
+
+  if (AllocTransientOp Op = dyn_cast<AllocTransientOp>(op))
+    return translateAllocTransientToSDFG(Op, jemit);
+
+  if (GetAccessOp Op = dyn_cast<GetAccessOp>(op))
+    return translateGetAccessToSDFG(Op, jemit);
+
+  if (LoadOp Op = dyn_cast<LoadOp>(op))
+    return translateLoadToSDFG(Op, jemit);
+
+  if (StoreOp Op = dyn_cast<StoreOp>(op))
+    return translateStoreToSDFG(Op, jemit);
+
+  if (CopyOp Op = dyn_cast<CopyOp>(op))
+    return translateCopyToSDFG(Op, jemit);
+
+  if (MemletCastOp Op = dyn_cast<MemletCastOp>(op))
+    return translateMemletCastToSDFG(Op, jemit);
+
+  if (ViewCastOp Op = dyn_cast<ViewCastOp>(op))
+    return translateViewCastToSDFG(Op, jemit);
+
+  if (SubviewOp Op = dyn_cast<SubviewOp>(op))
+    return translateSubviewToSDFG(Op, jemit);
+
+  if (AllocStreamOp Op = dyn_cast<AllocStreamOp>(op))
+    return translateAllocStreamToSDFG(Op, jemit);
+
+  if (AllocTransientStreamOp Op = dyn_cast<AllocTransientStreamOp>(op))
+    return translateAllocTransientStreamToSDFG(Op, jemit);
+
+  if (StreamPopOp Op = dyn_cast<StreamPopOp>(op))
+    return translateStreamPopToSDFG(Op, jemit);
+
+  if (StreamPushOp Op = dyn_cast<StreamPushOp>(op))
+    return translateStreamPushToSDFG(Op, jemit);
+
+  if (StreamLengthOp Op = dyn_cast<StreamLengthOp>(op))
+    return translateStreamLengthToSDFG(Op, jemit);
+
+  if (sdir::CallOp Op = dyn_cast<sdir::CallOp>(op))
+    return translateCallToSDFG(Op, jemit);
+
+  if (LibCallOp Op = dyn_cast<LibCallOp>(op))
+    return translateLibCallToSDFG(Op, jemit);
+
+  if (AllocSymbolOp Op = dyn_cast<AllocSymbolOp>(op))
+    return translateAllocSymbolToSDFG(Op, jemit);
+
+  if (SymOp Op = dyn_cast<SymOp>(op))
+    return translateSymbolExprToSDFG(Op, jemit);
+
+  // TODO: Implement ConstantOp & FuncOp
```

Comment:
Isn't this taken care of by your arith.constant handling?

Reply:
Indeed